### ISSUE TITLE
Omitting column names for INSERT statement gives error

### ISSUE THREAD

**Issue by breml** [sqlc-dev/sqlc#938]
Given a table:

```sql
CREATE TABLE status
(
  id integer,
  name varchar(100) NOT NULL,
  PRIMARY KEY(id)
);
```

the following INSERT query is valid in PostgreSQL:

```sql
INSERT INTO status VALUES (1, 'initial state');
```

Therefore I expect the following definition to be recognized by `sqlc`:

```sql
-- InsertStatus :exec
INSERT INTO status VALUES ($1, $2);
```

But `sqlc` gives me the following error:

```none
queries.sql:2:1: INSERT has more expressions than target columns
```

**Comment by breml** [comment]
For such an `INSERT` query I would expect the generated query function to take the `Status` struct as argument instead of generating an additional `InsertStatusParams` struct.

The same is true, if I define an `INSERT` query, which lists all columns of the target table.

**Comment by breml** [comment]
Related discussion: #587

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/insert_values_only/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/ca43b859e8ab621ab99cc9bbfac5798ce92347d8de49ca1d2a35000f8b05895e)

**Comment by Dan-Ailenei** [comment]
Hello! It isa bit weird that no matter if the fields are nullable or not, the generated code generates only pgx nullable types. Makes it a bit annoying to work with
